src/common/histogram.cpp

```diff
@@ -0,0 +1,50 @@
+#include "histogram.hpp"
+
+namespace shar {
+
+Histogram::Histogram() 
+  : m_family(nullptr)
+  , m_histogram(nullptr)
+{}
+
+Histogram::Histogram(const MetricsContext& context, std::shared_ptr<prometheus::Registry> registry, 
+                                                                       std::vector<double>& bounds) {
```

Comment:
`context` - pass by value and move from it, no need for extra copies.
`registry` - pass by value
`bounds` - `std::move` or pass by const reference. (first one is preferred, if `Add` accepts vector by value)